trl/trainer/iterative_trainer.py

```diff
@@ -0,0 +1,333 @@
+
+        else:
+            input_data = self.data_collator(
+                [{"input_ids": ids, "attention_mask": att} for ids, att in zip(input_ids, attention_mask)]
+            ).to(self.model.device)
+
+        return input_data
+
+    def compute_loss(self, model, inputs, return_outputs=False):
+        """
+        Loss is computed as in the HuggingFace Trainer.
+
+        Subclass and override for custom behavior.
+        """
+
+        outputs = model(**inputs)
+
+        if isinstance(outputs, dict) and "loss" not in outputs:
+            raise ValueError(
+                "The model did not return a loss from the inputs, only the following keys: "
+                f"{','.join(outputs.keys())}. For reference, the inputs it received are {','.join(inputs.keys())}."
+            )
+        # We don't use .loss here since the model may return tuples instead of ModelOutput.
+        loss = outputs["loss"] if isinstance(outputs, dict) else outputs[0]
+
+        return (loss, outputs) if return_outputs else loss
+
+    @staticmethod
+    def _step_safety_checker(
+        input_ids: List[torch.LongTensor], attention_mask: List[torch.LongTensor], labels: List[torch.LongTensor]
+    ):
+        """
+        Check if the input data is valid for training.
+
+        Args:
+            input_ids (List[`torch.LongTensor`]):
+                List of tensors containing the input_ids
+            attention_mask (List[`torch.LongTensor`]):
+                List of tensors containing the attention_mask
+            labels (List[`torch.FloatTensor`]):
+                List of tensors containing the labels
+        Returns:
+            `tuple`: The input processed data.
+        """
+        if attention_mask is None:
+            for name, tensor_list in zip(["input_ids", "labels"], [input_ids, labels]):
+                if not isinstance(tensor_list, list):
+                    raise ValueError(f"{name} must be a list of tensors - got {type(tensor_list)}")
+                if not isinstance(tensor_list[0], torch.Tensor):
+                    raise ValueError(f"Elements in {name} must be tensors - got {type(tensor_list[0])}")
+        else:
+            for name, tensor_list in zip(
+                ["input_ids", "attention_mask", "labels"], [input_ids, attention_mask, labels]
+            ):
+                if not isinstance(tensor_list, list):
+                    raise ValueError(f"{name} must be a list of tensors - got {type(tensor_list)}")
+                if not isinstance(tensor_list[0], torch.Tensor):
+                    raise ValueError(f"Elements in {name} must be tensors - got {type(tensor_list[0])}")
+
+        return input_ids, attention_mask, labels
+
+    @PPODecorators.empty_cuda_cache()
+    def step(
+        self,
+        input_ids: List[torch.LongTensor],
+        attention_mask: Optional[List[torch.LongTensor]],
+        labels: Optional[List[torch.LongTensor]],
```

Comment:
There is some design decision we have to take here. The `SFTTrainer` takes texts as input whereas the `PPOTrainer.step` uses tokenized inputs. I tend think that texts would be more natural but a compromise could be that we have an additional `texts` kwarg and make all args kwargs. Then the user can optionally pass tokenized or text inputs. What do you think?